zerver/lib/retention.py

```diff
@@ -0,0 +1,25 @@
+from datetime import datetime, timedelta
+from zerver.models import Realm, Message
+
+from typing import Optional, Generator
+
+def get_realm_expired_messages(realm):
+    # type(Realm) -> Optional(Dict[str, Any])
+    # Get expired_messages for Realm.
+    expired_date = datetime.now() - timedelta(days=realm.message_retention_days)
```

Comment:
We should use `timezone.now()` to match how we set message.pub_date (this will create a timezone-aware datetime properly).